File: kotlin/1929-concatenation-of-array.kt
/*
* Can be generalized to repeat(x) easier
*/
class Solution {
    fun getConcatenation(nums: IntArray): IntArray {
        val ans = LinkedList<Int>()
        repeat(2) {
            for(num in nums)
                ans.addLast(num)
        }
        return ans.toIntArray()
    }
}

/*
* concrete solution
*/
class Solution {
    fun getConcatenation(nums: IntArray): IntArray {
        val ans = IntArray(nums.size * 2)
        for(i in 0..nums.lastIndex) {
            ans[i] = nums[i]
            ans[i + nums.size] = nums[i]
        }
        return ans
    }
}

File: rust/1929-concatenation-of-array.rs
impl Solution {
    pub fn get_concatenation(nums: Vec<i32>) -> Vec<i32> {
        let n = nums.len();
        let mut ans = vec![0; 2 * n];

        for i in 0..nums.len() {
            ans[i] = nums[i];
            ans[i + n] = nums[i];
        }

        ans
    }
}
